### ISSUE TITLE
avr_loss = nan when training lokr for wan2.1 14b t2v task.

### ISSUE THREAD

**Issue by haoxu2000** [kohya-ss/musubi-tuner#394]
hello, I'm trying to train a lokr for wan2.1 14b(t2v task), but the loss goes nan. so I guess something went wrong here. only images used in the dataset.

INFO:musubi_tuner.modules.fp8_optimization_utils:Number of optimized Linear layers: 400
INFO:musubi_tuner.modules.fp8_optimization_utils:Number of monkey-patched Linear layers: 400
INFO:musubi_tuner.wan.modules.model:Moving weights to cuda
INFO:musubi_tuner.wan.modules.model:Loaded DiT model from /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_fp16.safetensors, info=<All keys matched successfully>
import network module: lycoris.kohya
2025-07-29 18:32:25|[LyCORIS]-INFO: Full matrix mode for LoKr is enabled
2025-07-29 18:32:25|[LyCORIS]-INFO: Using rank adaptation algo: lokr
2025-07-29 18:32:25|[LyCORIS]-INFO: Disable conv layer
2025-07-29 18:32:25|[LyCORIS]-INFO: Use Dropout value: 0.0
2025-07-29 18:32:25|[LyCORIS]-INFO: Create LyCORIS Module
2025-07-29 18:32:25|[LyCORIS]-INFO: create LyCORIS for U-Net: 400 modules.
2025-07-29 18:32:25|[LyCORIS]-INFO: module type table: {'LokrModule': 400}
2025-07-29 18:32:25|[LyCORIS]-INFO: enable LyCORIS for U-Net
WanModel: Gradient checkpointing enabled.
prepare optimizer, data loader etc.
INFO:musubi_tuner.hv_train_network:use 8-bit AdamW optimizer | {'weight_decay': 0.01}
override steps. steps for 200 epochs is / 指定エポックまでのステップ数: 10800
running training / 学習開始
  num train items / 学習画像、動画数: 107
  num batches per epoch / 1epochのバッチ数: 54
  num epochs / epoch数: 200
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 10800
INFO:musubi_tuner.hv_train_network:set DiT model name for metadata: /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_fp16.safetensors
INFO:musubi_tuner.hv_train_network:set VAE model name for metadata: /root/autodl-tmp/ComfyUI/models/vae/split_files/vae/wan_2.1_vae.safetensors
steps:   0%|                                                                                                                                                                                                            | 0/10800 [00:00<?, ?it/s]INFO:musubi_tuner.hv_train_network:DiT dtype: torch.float32, device: cuda:0

epoch 1/200
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
steps:   0%|▏                                                                                                                                                                                 | 10/10800 [00:59<17:48:32,  5.94s/it, avr_loss=nan]


training config :
dit = "/root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_fp16.safetensors"     
vae = "/root/autodl-tmp/ComfyUI/models/vae/split_files/vae/wan_2.1_vae.safetensors"   
t5 = "/root/autodl-tmp/ComfyUI/models/text_encoders/models_t5_umt5-xxl-enc-bf16.pth"
clip = "/root/autodl-tmp/ComfyUI/models/text_encoders/models_clip_open-clip-xlm-roberta-large-vit-huge-14.pth"
fp8_base = true 
fp8_scaled = true 
mixed_precision = "fp16"

dataset_config = "/root/autodl-tmp/kohya_training/wan2.1/config/wan_dataset_config.toml"
resolution = [768]

task = "t2v-14B"
max_train_epochs = 200 
gradient_checkpointing = true 
gradient_accumulation_steps = 1
guidance_scale = 1.0
seed = 1026

timestep_sampling = "shift"
discrete_flow_shfit = 3.0

lr = "2e-4"
lr_scheduler = "constant_with_warmup"
lr_warmup_steps = 100

optimizer_type = "adamw8bit"
optimizer_args =  ["weight_decay=0.01"]   

network_module="lycoris.kohya"
network_dim = 8
network_alpha = 1
network_args = [ "algo=lokr", "preset=attn-mlp", "factor=8", "full_matrix=True" ]
network_train_unet_only = true

flash_attn = true  
split_attn = true 
vae_cache_cpu = true 
max_data_loader_n_workers = 2 
persistent_data_loader_workers = true

**Comment by kohya-ss** [comment]
The calculations in LoKr may be unstable. Could you try changing the mixed precision from fp16 to bf16?

**Comment by haoxu2000** [comment]
thanks for your reply !

after changing the bf16 base model from (https://huggingface.co/Comfy-Org/Wan_2.1_ComfyUI_repackaged/tree/main/split_files/diffusion_models) and mixed_precision to bf16. the loss is quite high at 1.4.

is there any solution for that?

INFO:musubi_tuner.wan.modules.model:Loaded DiT model from /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_bf16.safetensors, info=<All keys matched successfully>
import network module: lycoris.kohya
2025-07-29 21:42:49|[LyCORIS]-INFO: Full matrix mode for LoKr is enabled
2025-07-29 21:42:49|[LyCORIS]-INFO: Using rank adaptation algo: lokr
2025-07-29 21:42:49|[LyCORIS]-INFO: Disable conv layer
2025-07-29 21:42:49|[LyCORIS]-INFO: Use Dropout value: 0.0
2025-07-29 21:42:49|[LyCORIS]-INFO: Create LyCORIS Module
2025-07-29 21:42:49|[LyCORIS]-INFO: create LyCORIS for U-Net: 400 modules.
2025-07-29 21:42:49|[LyCORIS]-INFO: module type table: {'LokrModule': 400}
2025-07-29 21:42:49|[LyCORIS]-INFO: enable LyCORIS for U-Net
WanModel: Gradient checkpointing enabled.
prepare optimizer, data loader etc.
INFO:musubi_tuner.hv_train_network:use 8-bit AdamW optimizer | {'weight_decay': 0.01}
override steps. steps for 200 epochs is / 指定エポックまでのステップ数: 10800
running training / 学習開始
  num train items / 学習画像、動画数: 107
  num batches per epoch / 1epochのバッチ数: 54
  num epochs / epoch数: 200
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 10800
INFO:musubi_tuner.hv_train_network:set DiT model name for metadata: /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_bf16.safetensors
INFO:musubi_tuner.hv_train_network:set VAE model name for metadata: /root/autodl-tmp/ComfyUI/models/vae/split_files/vae/wan_2.1_vae.safetensors
steps:   0%|                                                                                                                                                                                                            | 0/10800 [00:00<?, ?it/s]INFO:musubi_tuner.hv_train_network:DiT dtype: torch.bfloat16, device: cuda:0

epoch 1/200
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
steps:   0%|▉                                                                                                                                                                                | 54/10800 [05:14<17:23:06,  5.82s/it, avr_loss=1.43]
epoch 2/200
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 1, epoch: 2
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 1, epoch: 2
steps:   1%|█▏                                                                                                                                                                               | 71/10800 [06:51<17:17:12,  5.80s/it, avr_loss=1.42]

**Comment by kohya-ss** [comment]
The loss depends on the dataset and the discrete flow shift, but it certainly seems a little high. What happens if you switch to LoRA? If the loss is lower with LoRA, there may be some problem with LoKr.

**Comment by haoxu2000** [comment]
the lora is fine. avg_loss is around 0.08.

trained with there setting, the remain config is the same:
network_module = "networks.lora_wan"
network_dim = 32
network_alpha = 32


INFO:musubi_tuner.hv_train_network:Loading DiT model from /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_bf16.safetensors
INFO:musubi_tuner.wan.modules.model:Creating WanModel
INFO:musubi_tuner.wan.modules.model:Loading DiT model from /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_bf16.safetensors, device=cpu, dtype=None
INFO:musubi_tuner.wan.modules.model:Optimizing model weights to fp8. This may take a while.
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 400/400 [00:22<00:00, 18.03it/s]
INFO:musubi_tuner.modules.fp8_optimization_utils:Number of optimized Linear layers: 400
INFO:musubi_tuner.modules.fp8_optimization_utils:Number of monkey-patched Linear layers: 400
INFO:musubi_tuner.wan.modules.model:Moving weights to cuda
INFO:musubi_tuner.wan.modules.model:Loaded DiT model from /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_bf16.safetensors, info=<All keys matched successfully>
import network module: networks.lora_wan
INFO:musubi_tuner.networks.lora:create LoRA network. base dim (rank): 32, alpha: 32
INFO:musubi_tuner.networks.lora:neuron dropout: p=None, rank dropout: p=None, module dropout: p=None
INFO:musubi_tuner.networks.lora:create LoRA for U-Net/DiT: 400 modules.
INFO:musubi_tuner.networks.lora:LoRA+ UNet LR Ratio: 4.0
INFO:musubi_tuner.networks.lora:enable LoRA for U-Net: 400 modules
WanModel: Gradient checkpointing enabled.
prepare optimizer, data loader etc.
INFO:musubi_tuner.hv_train_network:use 8-bit AdamW optimizer | {'weight_decay': 0.01}
override steps. steps for 200 epochs is / 指定エポックまでのステップ数: 10800
running training / 学習開始
  num train items / 学習画像、動画数: 107
  num batches per epoch / 1epochのバッチ数: 54
  num epochs / epoch数: 200
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 10800
INFO:musubi_tuner.hv_train_network:set DiT model name for metadata: /root/autodl-tmp/ComfyUI/models/diffusion_models/split_files/diffusion_models/wan2.1_t2v_14B_bf16.safetensors
INFO:musubi_tuner.hv_train_network:set VAE model name for metadata: /root/autodl-tmp/ComfyUI/models/vae/split_files/vae/wan_2.1_vae.safetensors
steps:   0%|                                                                                                                                                                                                            | 0/10800 [00:00<?, ?it/s]INFO:musubi_tuner.hv_train_network:DiT dtype: torch.bfloat16, device: cuda:0

epoch 1/200
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:musubi_tuner.dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
steps:   0%|▋                                                                                                                                                                              | 42/10800 [03:02<12:57:41,  4.34s/it, avr_loss=0.0833]

**Comment by red-kestrel** [comment]
The problem should not persist if you don't do fp8_scaled, can you try without it?

From what I have tried, doras and lokr from lycoris don't seem to react well with fp8_scaled.

**Comment by kohya-ss** [comment]
I'm not familiar with LoKr, but you might try lowering the learning rate or changing the optimizer to something like AdamW.

**Comment by haoxu2000** [comment]
thanks! @red-kestrel 
after deleting fp8_scaled, the loss for wan2.1 works fine as normal lora.
